sklearn/ensemble/iforest.py

```diff
@@ -250,17 +250,28 @@ def decision_function(self, X):
         """
         # code structure from ForestClassifier/predict_proba
         # Check data
-        X = self.estimators_[0]._validate_X_predict(X, check_input=True)
+        X = check_array(X, accept_sparse='csr')
         n_samples = X.shape[0]
 
         n_samples_leaf = np.zeros((n_samples, self.n_estimators), order="f")
         depths = np.zeros((n_samples, self.n_estimators), order="f")
 
-        for i, tree in enumerate(self.estimators_):
-            leaves_index = tree.apply(X)
-            node_indicator = tree.decision_path(X)
+        if self._max_features == X.shape[1]:
+            random_projection = True
```

Comment:
@ngoix I find less code more simple ;) But I'm fine with keeping the variable and just renaming it. Then LGTM